packages/components/src/aspect-ratio/component.tsx

```diff
@@ -0,0 +1,70 @@
+/**
+ * External dependencies
+ */
+import { css } from '@emotion/react';
+
+/**
+ * WordPress dependencies
+ */
+import { cloneElement } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import {
+	contextConnect,
+	useContextSystem,
+	PolymorphicComponentProps,
+} from '../ui/context';
+import type { AspectRatioProps } from './types';
+import * as styles from './styles';
+import { useCx } from '../utils/hooks';
+import { getValidChildren } from '../ui/utils/get-valid-children';
+
+const { AspectRatioResizer, AspectRatioView } = styles;
+
+function AspectRatio(
+	props: PolymorphicComponentProps< AspectRatioProps, 'div' >,
+	forwardedRef: import('react').Ref< any >
+) {
+	const {
+		children,
+		className,
+		ratio = 1,
+		width,
+		...otherProps
+	} = useContextSystem( props, 'AspectRatio' );
+	const cx = useCx();
+
+	/**
+	 * Noting that only the first valid ReactElement will be actually
+	 * rendered. Other children (if any) are ignored.
+	 */
+	const [ child ] = getValidChildren( children );
+	const clonedChild =
+		child &&
```

Comment:
I could use some help here with types compatibility. I couldn't figure out what the appropriate change is here.

Reply:
I think you still have to check `isValidElement`? What's the error you're seeing?

Reply:
`getValidChildren` util uses `isValidElement`.

It's a type mismatch.

Reply:
The issue is the `isValidElement` will allow you to cast the child to `ReactElement` which lets you access props etc. But `getValidChildren` only returns ReactNodeArray because it also handles the case where the children are `string`s for example, so the type has to be wider.

Maybe `getValidChildren` is unnecessary here and we can call `Children.toArray(children)[0]` directly and then check it with `isValidElement`. `getValidChildren` seems semi-useless in this regard. In `HStack` component we end up force casting the children to `ReactElement` using `@ts-ignore`.